packages/manager/apps/web/client/app/hosting/multisite/hosting-multisite.service.js

```diff
@@ -337,24 +337,28 @@ angular.module('services').service(
     }
 
     /**
-     *
-     * @param {string} serviceName
      * @param {string} domain
      * @param {string} subDomain
      * @param {boolean} wwwNeeded
      */
-    getExistingConfiguration(serviceName, domain, subDomain, wwwNeeded) {
-      return this.$http
-        .get(
-          `${this.aapiHostingPath}/${serviceName}/domains/${domain}/configuration`,
-          {
-            params: {
-              domainName: subDomain,
-              wwwNeeded,
-            },
-          },
-        )
-        .then((response) => response.data);
+    getExistingConfiguration(domain, subDomain, wwwNeeded) {
+      return this.iceberg(`/domain/zone/${domain}/record`)
+        .query()
+        .expand('CachedObjectList-Pages')
+        .addFilter('fieldType', 'eq', 'A')
+        .execute()
+        .$promise.then(({ data }) =>
+          data.filter((record) => {
+            if (wwwNeeded) {
+              return (
+                record.subDomain === subDomain ||
+                record.subDomain ===
+                  (subDomain !== '' ? ['www', subDomain].join('.') : 'www')
+              );
+            }
+            return record.subDomain === subDomain;
```

Comment:
```suggestion
            return record.subDomain === subDomain ||  (wwwNeeded && record.subDomain === ['www', subDomain].filter(Boolean).join('.')) ;
```

It might do the same thing not sure if it's more readable tho